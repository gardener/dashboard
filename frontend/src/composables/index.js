//
// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

export { useApi } from './useApi'
export { useAuthn } from './useAuthn'
export { useLogger } from './useLogger'
export { useNav } from './useNav'
export { useTheme } from './useTheme'
export { useToken } from './useToken'
export { useUser } from './useUser'
