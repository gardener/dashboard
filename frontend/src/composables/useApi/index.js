//
// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

import { markRaw } from 'vue'

import api from './api'
import { registry } from './fetch'

export const useApi = () => {
  return markRaw(api)
}

export const useInterceptors = () => {
  return markRaw(registry)
}
