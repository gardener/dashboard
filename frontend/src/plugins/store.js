//
// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

import pinia from '@/store'

export default {
  install (app) {
    app.use(pinia)
  },
}
