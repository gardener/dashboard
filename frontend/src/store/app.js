//
// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

import { defineStore } from 'pinia'
import { ref } from 'vue'

import moment from '@/utils/moment'
import { errorDetailsFromError } from '@/utils/error'

export const useAppStore = defineStore('app', () => {
  const ready = ref(false)
  const sidebar = ref(true)
  const redirectPath = ref(null)
  const loading = ref(false)
  const alert = ref(null)
  const location = ref(moment.tz.guess())
  const timezone = ref(moment().format('Z'))
  const focusedElementId = ref(null)
  const splitpaneResize = ref(null)
  const fromRoute = ref(null)
  const routerError = ref(null)

  function updateSplitpaneResize () {
    splitpaneResize.value = new Date()
  }

  function setAlert (value) {
    alert.value = value
  }

  function setAlertWithType (type, value) {
    const alert = {
      type,
    }
    if (typeof value === 'string') {
      alert.message = value
    } else if (value) {
      const { message = '', title } = value
      alert.message = value.response
        ? errorDetailsFromError(value).detailedMessage
        : message
      if (title) {
        alert.title = title
      }
    } else {
      alert.message = ''
    }
    setAlert(alert)
  }

  function setError (value) {
    setAlertWithType('error', value)
  }

  function setSuccess (value) {
    setAlertWithType('success', value)
  }

  function setRouterError (value) {
    routerError.value = value
  }

  return {
    ready,
    sidebar,
    redirectPath,
    loading,
    alert,
    location,
    timezone,
    focusedElementId,
    splitpaneResize,
    fromRoute,
    routerError,
    updateSplitpaneResize,
    setAlert,
    setError,
    setSuccess,
    setRouterError,
  }
})
