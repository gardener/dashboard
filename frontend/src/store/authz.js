//
// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

import {
  defineStore,
  acceptHMRUpdate,
} from 'pinia'
import {
  ref,
  computed,
} from 'vue'

import { useApi } from '@/composables/useApi'

import { canI } from '@/utils'

import { useConfigStore } from './config'
import { useAuthnStore } from './authn'
import { useProjectStore } from './project'

export const useAuthzStore = defineStore('authz', () => {
  const api = useApi()
  const authnStore = useAuthnStore()
  const configStore = useConfigStore()
  const projectStore = useProjectStore()

  const spec = ref(null)
  const status = ref(null)

  const namespace = computed(() => {
    return spec.value?.namespace
  })

  const canCreateTerminals = computed(() => {
    return canI(status.value, 'create', 'dashboard.gardener.cloud', 'terminals')
  })

  const canCreateShoots = computed(() => {
    return canI(status.value, 'create', 'core.gardener.cloud', 'shoots')
  })

  const canPatchShoots = computed(() => {
    return canI(status.value, 'patch', 'core.gardener.cloud', 'shoots')
  })

  const canDeleteShoots = computed(() => {
    return canI(status.value, 'delete', 'core.gardener.cloud', 'shoots')
  })

  const canPatchShootsBinding = computed(() => {
    return canI(status.value, 'patch', 'core.gardener.cloud', 'shoots/binding')
  })

  const canGetSecrets = computed(() => {
    return canI(status.value, 'list', '', 'secrets')
  })

  const canCreateSecrets = computed(() => {
    return canI(status.value, 'create', '', 'secrets')
  })

  const canCreateShootsAdminkubeconfig = computed(() => {
    return canI(status.value, 'create', 'core.gardener.cloud', 'shoots/adminkubeconfig')
  })

  const canCreateShootsViewerkubeconfig = computed(() => {
    return canI(status.value, 'create', 'core.gardener.cloud', 'shoots/viewerkubeconfig')
  })

  const canPatchSecrets = computed(() => {
    return canI(status.value, 'patch', '', 'secrets')
  })

  const canDeleteSecrets = computed(() => {
    return canI(status.value, 'delete', '', 'secrets')
  })

  const canCreateTokenRequest = computed(() => {
    return canI(status.value, 'create', '', 'serviceaccounts/token')
  })

  const canCreateServiceAccounts = computed(() => {
    return canI(status.value, 'create', '', 'serviceaccounts')
  })

  const canPatchServiceAccounts = computed(() => {
    return canI(status.value, 'patch', '', 'serviceaccounts')
  })

  const canDeleteServiceAccounts = computed(() => {
    return canI(status.value, 'delete', '', 'serviceaccounts')
  })

  const canCreateProject = computed(() => {
    return canI(status.value, 'create', 'core.gardener.cloud', 'projects')
  })

  const canPatchProject = computed(() => {
    return canI(status.value, 'patch', 'core.gardener.cloud', 'projects', projectStore.projectName)
  })

  const canManageMembers = computed(() => {
    return canI(status.value, 'manage-members', 'core.gardener.cloud', 'projects', projectStore.projectName)
  })

  const canManageServiceAccountMembers = computed(() => {
    return canPatchProject.value || canManageMembers.value
  })

  const canDeleteProject = computed(() => {
    return canI(status.value, 'delete', 'core.gardener.cloud', 'projects', projectStore.projectName)
  })

  const canGetProjectTerminalShortcuts = computed(() => {
    return canGetSecrets.value
  })

  const canUseProjectTerminalShortcuts = computed(() => {
    return configStore.isProjectTerminalShortcutsEnabled &&
      canGetProjectTerminalShortcuts.value &&
      canCreateTerminals.value
  })

  const hasGardenTerminalAccess = computed(() => {
    return configStore.isTerminalEnabled &&
    canCreateTerminals.value &&
    canPatchServiceAccounts.value &&
    canCreateServiceAccounts.value
  })

  const hasControlPlaneTerminalAccess = computed(() => {
    return configStore.isTerminalEnabled &&
      canCreateTerminals.value &&
      authnStore.isAdmin
  })

  const hasShootTerminalAccess = computed(() => {
    return configStore.isTerminalEnabled &&
      canCreateTerminals.value
  })

  // reuse function not exported
  async function getRules (namespace) {
    const body = { namespace }
    const response = await api.getSubjectRules(body)
    status.value = response.data
  }

  async function fetchRules (namespace) {
    /**
     * The value of `spec.value?.namespace` is:
     * - undefined if no rules have been fetched yet
     * - null if only cluster scoped rules have been fetched
     * - a non-empty string if both cluster scoped rules and the rules for the namespace have been fetched
     */
    if (!namespace) {
      namespace = null
    }
    if (spec.value?.namespace !== namespace) {
      await getRules(namespace)
      this.setNamespace(namespace)
    }
  }

  function refreshRules () {
    return getRules(spec.value?.namespace)
  }

  function setNamespace (namespace) {
    spec.value = { namespace }
  }

  function $reset () {
    spec.value = null
    status.value = null
  }

  return {
    namespace,
    setNamespace,
    canCreateTerminals,
    canCreateShoots,
    canPatchShoots,
    canDeleteShoots,
    canPatchShootsBinding,
    canGetSecrets,
    canCreateSecrets,
    canCreateShootsAdminkubeconfig,
    canCreateShootsViewerkubeconfig,
    canPatchSecrets,
    canDeleteSecrets,
    canCreateTokenRequest,
    canCreateServiceAccounts,
    canPatchServiceAccounts,
    canDeleteServiceAccounts,
    canCreateProject,
    canPatchProject,
    canDeleteProject,
    canManageMembers,
    canManageServiceAccountMembers,
    canGetProjectTerminalShortcuts,
    canUseProjectTerminalShortcuts,
    hasGardenTerminalAccess,
    hasControlPlaneTerminalAccess,
    hasShootTerminalAccess,
    fetchRules,
    refreshRules,
    $reset,
  }
})

if (import.meta.hot) {
  import.meta.hot.accept(acceptHMRUpdate(useAuthzStore, import.meta.hot))
}
