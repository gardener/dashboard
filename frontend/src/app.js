//
// Copyright (c) 2020 by SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

import Vue from 'vue'
import store from '@/store'
import createRouter from '@/router'

Vue.config.productionTip = false

const App = Vue.extend({
  name: 'app',
  created () {
    // provide the keyboard events for dialogs. Dialogs can't catch keyboard events
    // if any input element of the dialog didn't have the focus.
    window.addEventListener('keyup', ({ keyCode }) => {
      if (keyCode === 27) {
        this.$bus.$emit('esc-pressed')
      }
    })
  },
  render (createElement) {
    return createElement('router-view')
  }
})

function createApp (vuetify) {
  return new Vue({
    vuetify,
    store,
    router: createRouter(store),
    render (createElement) {
      return createElement(App)
    }
  })
}

export { App, createApp }

export default createApp
