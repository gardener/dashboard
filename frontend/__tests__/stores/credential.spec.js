//
// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

import {
  setActivePinia,
  createPinia,
} from 'pinia'

import { useAppStore } from '@/store/app'
import { useAuthzStore } from '@/store/authz'
import { useCloudProfileStore } from '@/store/cloudProfile'
import { useCredentialStore } from '@/store/credential'
import { useGardenerExtensionStore } from '@/store/gardenerExtension'

import { useApi } from '@/composables/useApi'

import find from 'lodash/find'

const testNamespace = 'garden-test'
const awsSecretBindingName = 'aws-secretbinding'
const awsSecretName = 'aws-secret'
const awsTrialSecretBindingName = 'aws-trial-secretbinding'
const azureSecretBindingName = 'azure-secretbinding'

describe('stores', () => {
  describe('credential', () => {
    let api
    let appStore
    let authzStore
    let cloudProfileStore
    let gardenerExtensionStore
    let credentialStore

    beforeEach(async () => {
      setActivePinia(createPinia())
      api = useApi()
      vi.spyOn(api, 'getCloudProviderCredentials').mockImplementation(namespace => {
        if (namespace !== testNamespace) {
          throw new Error('Unauthorized')
        }
        return {
          data: fixtures.credentials,
        }
      })
      vi.spyOn(api, 'updateCloudProviderCredential').mockImplementation(({ secretBinding, secret }) => {
        return {
          data: {
            secretBinding,
            secret,
          },
        }
      })
      vi.spyOn(api, 'createCloudProviderCredential').mockImplementation(({ secretBinding, secret }) => {
        return {
          data: {
            secretBinding,
            secret,
          },
        }
      })
      vi.spyOn(api, 'deleteCloudProviderCredential').mockReturnValue()
      appStore = useAppStore()
      vi.spyOn(appStore, 'setSuccess')
      authzStore = useAuthzStore()
      authzStore.setNamespace(testNamespace)
      cloudProfileStore = useCloudProfileStore()
      cloudProfileStore.list = fixtures.cloudprofiles
      gardenerExtensionStore = useGardenerExtensionStore()
      gardenerExtensionStore.list = fixtures.gardenerExtensions
      credentialStore = useCredentialStore()
      credentialStore._setCredentials(fixtures.credentials)
    })

    afterEach(() => {
      appStore.setSuccess.mockClear()
      api.getCloudProviderCredentials.mockClear()
      api.createCloudProviderCredential.mockClear()
      api.updateCloudProviderCredential.mockClear()
      api.deleteCloudProviderCredential.mockClear()
    })

    it('should create a new credential store', () => {
      expect(cloudProfileStore.sortedProviderTypeList).toEqual(
        expect.arrayContaining(['aws', 'azure']),
      )
      expect(gardenerExtensionStore.dnsProviderTypes).toEqual(
        expect.arrayContaining(['aws-route53', 'azure-dns']),
      )
    })

    it('should return secretBindingList with resolved secret', () => {
      expect(credentialStore.secretBindingList.length).toBeGreaterThan(0)
      const awsSecretBinding = find(credentialStore.secretBindingList, { metadata: { name: awsSecretBindingName } })
      expect(awsSecretBinding.secretRef.name).toBe(awsSecretBinding._secret.metadata.name)
      expect(awsSecretBinding.secretRef.namespace).toBe(awsSecretBinding._secret.metadata.namespace)
    })

    it('should return secretBindingList with trial quota and secret from other namespace', () => {
      const awsTrialSecretBinding = find(credentialStore.secretBindingList, { metadata: { name: awsTrialSecretBindingName } })
      expect(awsTrialSecretBinding._secret).toBeUndefined()
      expect(awsTrialSecretBinding.secretRef.namespace).toBe(awsTrialSecretBinding._quotas[0].metadata.namespace)
    })

    it('should return secretBindingList with multiple quotas', () => {
      const azureSecretBinding = find(credentialStore.secretBindingList, { metadata: { name: azureSecretBindingName } })
      expect(azureSecretBinding._secret).toMatchSnapshot()
      expect(azureSecretBinding._quotas.length).toBe(2)
      const azureQuotaRef2 = azureSecretBinding.quotas[1]
      const azureQuota = find(credentialStore.quotaList, { metadata: azureQuotaRef2 })
      expect(azureQuota).toMatchSnapshot()
    })

    it('should return infrastructureSecretBindingsList', () => {
      expect(credentialStore.infrastructureSecretBindingsList.length).toBeGreaterThan(0)
      expect(credentialStore.secretBindingList.length).toBeGreaterThan(credentialStore.infrastructureSecretBindingsList.length)
      expect(cloudProfileStore.sortedProviderTypeList).toMatchSnapshot()
    })

    it('should return dnsSecretBindingsList', () => {
      expect(credentialStore.dnsSecretBindingsList.length).toBeGreaterThan(0)
      expect(credentialStore.secretBindingList.length).toBeGreaterThan(credentialStore.dnsSecretBindingsList.length)
      expect(gardenerExtensionStore.dnsProviderTypes).toMatchSnapshot()
    })

    it('should return secret', () => {
      expect(credentialStore.getSecret({ namespace: testNamespace, name: awsSecretName })).toMatchSnapshot()
    })

    it('should return secretBinding', () => {
      expect(credentialStore.getSecretBinding({ namespace: testNamespace, name: awsSecretBindingName })).toMatchSnapshot()
    })

    it('should fetchCredentials', async () => {
      credentialStore.$reset()
      expect(credentialStore.secretBindingList.length).toBe(0)
      await credentialStore.fetchCredentials()
      expect(api.getCloudProviderCredentials).toBeCalledTimes(1)
      expect(api.getCloudProviderCredentials).toBeCalledWith(testNamespace)
      expect(credentialStore.secretBindingList.length).toBeGreaterThan(0)
    })

    it('should updateCredential', async () => {
      let awsSecretBinding = find(credentialStore.secretBindingList, { metadata: { name: awsSecretBindingName } })
      const secret = {
        ...awsSecretBinding._secret,
        data: {
          newSecret2: 'dummy-data',
        },
      }
      await credentialStore.updateCredential({ secretBinding: awsSecretBinding, secret })

      expect(api.updateCloudProviderCredential).toBeCalledTimes(1)
      expect(api.updateCloudProviderCredential).toBeCalledWith({ secretBinding: awsSecretBinding, secret })
      awsSecretBinding = find(credentialStore.secretBindingList, { metadata: { name: awsSecretBindingName } })
      expect(awsSecretBinding._secret.data).toEqual({ newSecret2: 'dummy-data' })
    })

    it('should createCredential', async () => {
      const secretBinding = {
        apiVersion: 'core.gardener.cloud/v1alpha1',
        kind: 'SecretBinding',
        metadata: {
          namespace: testNamespace,
          name: 'my-new-secret-binding',
        },
        provider: {
          type: 'aws',
        },
        secretRef: {
          namespace: testNamespace,
          name: 'my-new-secret',
        },
      }
      const secret = {
        apiVersion: 'v1',
        kind: 'Secret',
        type: 'Opaque',
        metadata: {
          name: 'my-new-secret',
          namespace: testNamespace,
        },
        data: {
          newSecret: 'dummy-data',
        },
      }

      await credentialStore.createCredential({ secretBinding, secret })

      expect(api.createCloudProviderCredential).toBeCalledTimes(1)
      expect(api.createCloudProviderCredential).toBeCalledWith({ secretBinding, secret })
      const newSecretBinding = find(credentialStore.secretBindingList, { metadata: secretBinding.metadata })
      expect(newSecretBinding.metadata.namespace).toEqual(testNamespace)
      expect(newSecretBinding.provider.type).toEqual('aws')
      expect(newSecretBinding._secret.data).toEqual({ newSecret: 'dummy-data' })
    })

    it('should not delete credential', async () => {
      const name = azureSecretBindingName
      const namespace = testNamespace

      await credentialStore.deleteCredential(name)

      expect(api.deleteCloudProviderCredential).toBeCalledTimes(1)
      expect(api.deleteCloudProviderCredential).toBeCalledWith({ name, namespace })
      expect(api.getCloudProviderCredentials).toBeCalledTimes(1)
    })

    it('store should be resetted in case of a fetch error', async () => {
      const namespace = 'invalid'
      authzStore.setNamespace(namespace)

      expect(credentialStore.secretBindingList.length).toBeGreaterThan(0)
      await expect(credentialStore.fetchCredentials()).rejects.toThrow(Error)
      expect(credentialStore.secretBindingList.length).toBe(0)
    })
    it('store should be resetted after setting new data', async () => {
      const namespace = 'invalid'
      authzStore.setNamespace(namespace)

      expect(credentialStore.secretBindingList.length).toBeGreaterThan(0)
      credentialStore._setCredentials({})
      expect(credentialStore.secretBindingList.length).toBe(0)
    })
  })
})
