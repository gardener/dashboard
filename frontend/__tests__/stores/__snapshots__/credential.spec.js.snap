// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`stores > credential > should deleteCredential secretbinding and referenced secret / quota 1`] = `
{
  "metadata": {
    "name": "azure-secretbinding",
    "namespace": "garden-test",
  },
  "provider": {
    "type": "azure",
  },
  "quotas": [
    {
      "name": "azure-foo-quota",
      "namespace": "garden-trial",
    },
    {
      "name": "azure-bar-quota",
      "namespace": "garden-test",
    },
  ],
  "secretRef": {
    "name": "azure-secret",
    "namespace": "garden-test",
  },
}
`;

exports[`stores > credential > should deleteCredential secretbinding and referenced secret / quota 2`] = `
{
  "data": {
    "secret": "cw==",
  },
  "metadata": {
    "name": "azure-secret",
    "namespace": "garden-test",
  },
}
`;

exports[`stores > credential > should deleteCredential secretbinding and referenced secret / quota 3`] = `
{
  "metadata": {
    "name": "azure-foo-quota",
    "namespace": "garden-trial",
  },
}
`;

exports[`stores > credential > should deleteCredential secretbinding and referenced secret / quota 4`] = `
{
  "metadata": {
    "name": "azure-bar-quota",
    "namespace": "garden-test",
  },
}
`;

exports[`stores > credential > should not delete secret or quota if referenced by other SecretBinding 1`] = `
{
  "metadata": {
    "name": "azure-secretbinding",
    "namespace": "garden-test",
  },
  "provider": {
    "type": "azure",
  },
  "quotas": [
    {
      "name": "azure-foo-quota",
      "namespace": "garden-trial",
    },
    {
      "name": "azure-bar-quota",
      "namespace": "garden-test",
    },
  ],
  "secretRef": {
    "name": "azure-secret",
    "namespace": "garden-test",
  },
}
`;

exports[`stores > credential > should not delete secret or quota if referenced by other SecretBinding 2`] = `
{
  "data": {
    "secret": "cw==",
  },
  "metadata": {
    "name": "azure-secret",
    "namespace": "garden-test",
  },
}
`;

exports[`stores > credential > should not delete secret or quota if referenced by other SecretBinding 3`] = `
{
  "metadata": {
    "name": "azure-foo-quota",
    "namespace": "garden-trial",
  },
}
`;

exports[`stores > credential > should not delete secret or quota if referenced by other SecretBinding 4`] = `
{
  "metadata": {
    "name": "azure-bar-quota",
    "namespace": "garden-test",
  },
}
`;

exports[`stores > credential > should not delete secret or quota if referenced by other SecretBinding 5`] = `
{
  "data": {
    "secret": "cw==",
  },
  "metadata": {
    "name": "azure-secret",
    "namespace": "garden-test",
  },
}
`;

exports[`stores > credential > should not delete secret or quota if referenced by other SecretBinding 6`] = `
{
  "metadata": {
    "name": "azure-foo-quota",
    "namespace": "garden-trial",
  },
}
`;

exports[`stores > credential > should return dnsSecretBindingsList 1`] = `
[
  "aws-route53",
  "azure-dns",
  "azure-private-dns",
  "google-clouddns",
  "openstack-designate",
  "alicloud-dns",
  "infoblox-dns",
  "netlify-dns",
  "rfc2136",
]
`;

exports[`stores > credential > should return infrastructureSecretBindingsList 1`] = `
[
  "aws",
  "azure",
  "gcp",
  "openstack",
  "alicloud",
  "ironcore",
]
`;

exports[`stores > credential > should return secret 1`] = `
{
  "data": {
    "secret": "cw==",
  },
  "metadata": {
    "name": "aws-secret",
    "namespace": "garden-test",
  },
}
`;

exports[`stores > credential > should return secretBinding 1`] = `
{
  "metadata": {
    "name": "aws-secretbinding",
    "namespace": "garden-test",
  },
  "provider": {
    "type": "aws",
  },
  "secretRef": {
    "name": "aws-secret",
    "namespace": "garden-test",
  },
}
`;

exports[`stores > credential > should return secretBindingList with multiple quotas 1`] = `
{
  "data": {
    "secret": "cw==",
  },
  "metadata": {
    "name": "azure-secret",
    "namespace": "garden-test",
  },
}
`;

exports[`stores > credential > should return secretBindingList with multiple quotas 2`] = `
{
  "metadata": {
    "name": "azure-bar-quota",
    "namespace": "garden-test",
  },
}
`;
